lib/iris/cube.py

```diff
@@ -2263,8 +2180,18 @@ def new_cell_measure_dims(cm_):
             if ma.count_masked(data) == 0:
                 data = data.filled()
 
+        # XXX: Slicing a single item from a masked array that is masked,
+        #      results in numpy (v1.11.1) *always* returning a MaskedConstant
+        #      with a dtype of float64, regardless of the original masked
+        #      array dtype!
+        if isinstance(data, ma.core.MaskedConstant) and \
+                data.dtype != cube_data.dtype:
```

Comment:
I think this logic should be pushed down into the `DataManager.data` getter. That way, if there are other places where it occurs they will be handled in one place.